tools/py_common/btree_format.py

```diff
@@ -172,32 +188,98 @@ def __init__(self) -> None:
         self.unused = 0
 
     @staticmethod
-    def parse(b: binary_data.BinaryFile, disagg = False) -> 'BlockHeader':
+    def parse(b: binary_data.BinaryFile) -> 'BlockHeader':
         '''
         Parse a block header.
         '''
         # WT_BLOCK_HEADER in block.h (12 bytes)
         h = BlockHeader()
-        if disagg:
-            # Disagg sets additional fields.  If they are examined
-            # by non-disagg code, an exception will be thrown (by design).
-            h.disagg_magic = b.read_uint8()
-            h.disagg_version = b.read_uint8()
-            h.disagg_compatible_version = b.read_uint8()
-            h.disagg_header_size = b.read_uint8()
-            h.checksum = b.read_uint32()
-            h.disagg_previous_checksum = b.read_uint32()
-            h.disagg_reconciliation_id = b.read_uint8()
-            h.flags = b.read_uint8()
-            h.unused = int.from_bytes(b.read(2), byteorder='little')
-        else:
-            h.disk_size = b.read_uint32()
-            h.checksum = b.read_uint32()
-            h.flags = b.read_uint8()
-            h.unused = int.from_bytes(b.read(3), byteorder='little')
+        h.disk_size = b.read_uint32()
+        h.checksum = b.read_uint32()
+        h.flags = BlockFlags(b.read_uint8())
+        h.unused = int.from_bytes(b.read(3), byteorder='little')
         return h
+    
+    def __str__(self):
+        header_string = (
+            f"Block Header:\n"
+            f"  disk_size: {str(self.disk_size)}"
+            f"  checksum: {str(self.checksum)}"
+            f"  flags: {str(self.flags)}"
+        )
+        return header_string
+
+class BlockDisaggFlags(enum.IntFlag):
+    '''
+    Disagg block flags from block.h
+    '''
+    WT_BLOCK_DISAGG_DATA_CKSUM = 0x1
+    WT_BLOCK_DISAGG_ENCRYPTED = 0x2
+    WT_BLOCK_DISAGG_COMPRESSED = 0x4
+
+class BlockDisaggHeader(object):
+    '''
+    A block header (WT_BLOCK_DISAGG_HEADER).
+    '''
+    magic: int
+    version: int
+    compatible_version: int
+    header_size: int
+    checksum: int
+    previous_checksum: int
+    flags: BlockDisaggFlags
+    unused: int
 
+    # Block types (magic byte)
+    WT_BLOCK_DISAGG_MAGIC_BASE: typing.Final[int] = 0xdb
+    WT_BLOCK_DISAGG_MAGIC_DELTA: typing.Final[int] = 0xdd
 
+    def __init__(self) -> None:
+        '''
+        Initialize the instance with default values.
+        '''
+        self.magic = 0
+        self.version = 0
+        self.compatible_version = 0
+        self.header_size = 0
+        self.checksum = 0
+        self.previous_checksum = 0
+        self.flags = 0
+        self.unused = 0
+
+    @staticmethod
+    def parse(b: binary_data.BinaryFile, disagg = False) -> 'BlockDisaggHeader':
+        '''
+        Parse a block header.
+        '''
+        # WT_BLOCK_DISAGG_HEADER in block.h (16 bytes)
+        h = BlockDisaggHeader()
+        # Disagg sets additional fields.  If they are examined
```

Comment:
nit: `additional` to what?  Maybe "Disagg sets different fields from the standard block header."?

Reply:
That's a good point, I left this comment as it was in the code previously. But we can remove it now the classes are separated.

Reply:
I moved an explanation to the class description.